hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -62,6 +81,57 @@ interface HiveOperationsBase {
 
   String table();
 
+  String catalogName();
+
+  String entityType();
+
+  BaseMetastoreOperations.CommitStatus validateNewLocationAndReturnCommitStatus(
+      BaseMetadata metadata, String newMetadataLocation);
```

Comment:
we're only effectively passing properties in the implementation, so why does `BaseMetadata` help here?